src/inmanta/ast/__init__.py

```diff
@@ -143,6 +143,38 @@ def __eq__(self, other: object) -> bool:
         return False
 
 
+class AnchorTarget(Location):
+    __slots__ = ("start_char", "end_lnr", "end_char", "docstring")
```

Comment:
Why not `AnchorTarget(Range)`?

Reply:
because we don't always have a start_char, end_lnr and end_char which are not optional for Ranges

Reply:
Actually I think I'm already breaking compatibilty with previous versions of the extension. Some of it's behaviors depend on `if isinstance(t, Location)` or if `isinstance(t, Range)` that I had to change to `if isinstance(t, AnchorTarget)`. I noticed  this morning (it was breaking the code navigation), but I only realize now that this might be bad. I don't think there is another way around this other than to add the docstring on the location/range object which I don't like at all. 
So I think I could have a location field as i'm is already not backward compatible anymore :/

Reply:
do you mean where the anchormap is used?

Reply:
Yes. (When I asked the question I thought it was broader than that but I had another look at the diff and that's what it comes down to.)

Reply:
I think the anchormap is only used to created what is needed for the LS so I shouldn't break anything else.